packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeStablecoinPayment.sol

```diff
@@ -0,0 +1,289 @@
+
+    /// @notice Reverts if the airnode is not opted in
+    /// @param _airnode The airnode Address
+    modifier isOptedIn(address _airnode) {
+        require(airnodeSelfOptInStatus[_airnode], ERROR_AIRNODE_NOT_OPTED_IN);
+        _;
+    }
+
+    /// @notice Reverts if the erc20 is not supported by
+    /// default or by the airnode
+    /// @param _chainId The chainId
+    /// @param _stablecoin The address of the ERC20 stablecoin
+    /// @param _airnode The airnode Address
+    modifier isSupportedERC20(
+        address _stablecoin,
+        uint256 _chainId,
+        address _airnode
+    ) {
+        require(
+            defaultSupportedERC20[_chainId][_stablecoin] ||
+                airnodeSupportedERC20[_chainId][_airnode][_stablecoin],
+            "ERC20 not supported"
+        );
+        _;
+    }
+
+    /// @notice Called by a registry setter to set the address
+    /// of the AirnodeFeeRegistry contract
+    /// @param _airnodeFeeRegistry The address of the AirnodeFeeRegistry contract
+    function setAirnodeFeeRegistry(address _airnodeFeeRegistry)
+        external
+        override
+    {
+        require(_airnodeFeeRegistry != address(0), ERROR_ZERO_ADDRESS);
+        airnodeFeeRegistry = _airnodeFeeRegistry;
+        emit SetAirnodeFeeRegistry(_airnodeFeeRegistry, msg.sender);
+    }
+
+    /// @notice Called by an admin to set the default supported erc20 address on a chain
+    /// @param _chainId The chainId
+    /// @param _stablecoin The address of the stablecoin token contract
+    /// @param _status The supported status of the stablecoin
+    function setDefaultSupportedERC20(
+        uint256 _chainId,
+        address _stablecoin,
+        bool _status
+    ) external override {
+        defaultSupportedERC20[_chainId][_stablecoin] = _status;
+        emit SetDefaultSupportedERC20(
+            _chainId,
+            _stablecoin,
+            _status,
+            msg.sender
+        );
+    }
+
+    /// @notice Called by the airnode to set the supported erc20 address on a chain
+    /// @param _chainId The chainId
+    /// @param _airnode the address of the airnode
+    /// @param _stablecoin The address of the stablecoin token contract
+    /// @param _status The supported status of the stablecoin
+    function setAirnodeSupportedERC20(
+        uint256 _chainId,
+        address _airnode,
+        address _stablecoin,
+        bool _status
+    ) external override {
+        require(msg.sender == _airnode, ERROR_NOT_AIRNODE);
+        airnodeSupportedERC20[_chainId][_airnode][_stablecoin] = _status;
+        emit SetAirnodeSupportedERC20(
+            _chainId,
+            _airnode,
+            _stablecoin,
+            _status,
+            msg.sender
+        );
+    }
+
+    /// @notice Called by the airnode to set the opt status for itself
+    /// @param _airnode The airnode address
+    /// @param _status The Opted status for the airnode
+    function setSelfOptInStatus(address _airnode, bool _status)
```

Comment:
I don't like that stablecoin payments require a transaction from the node operator